### PR TITLE
RUBY-1793 add endpoints and tests

### PR DESCRIPTION
**Pull request by kaloyan-mm**
This PR adds missing endpoints and requests for sinatra and grape

### CODE REVIEW

**Review by hmcginnis — APPROVED** [review]

**Review by cshep-contrast — APPROVED** [review]

**Review by Teodor-Raychev — APPROVED** [review]

**Review by srushe — APPROVED** [review]

File: Gemfile.lock [hunk 1]
```diff
@@ -12,85 +12,88 @@ PATH
...
       sprockets-rails
+      webpacker
 
 GEM
   remote: https://rubygems.org/
   specs:
-    actioncable (7.0.3.1)
-      actionpack (= 7.0.3.1)
-      activesupport (= 7.0.3.1)
+    actioncable (7.0.4)
```
Comment (srushe) [inline Gemfile.lock:28]: 👏

File: app/controllers/vulneruby_engine/grape_controller.rb [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # frozen_string_literal: true
 
 require 'grape'
+require 'fileutils'
 
 module VulnerubyEngine
   # Base controller for the Grape framework mount
   class GrapeController < Grape::API
+    include FileUtils::Verbose
```
Comment (srushe) [inline app/controllers/vulneruby_engine/grape_controller.rb:9]: I've used FileUtils so many times and only now found out about this 😲

File: app/controllers/vulneruby_engine/insecure_algorithm_controller.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ module VulnerubyEngine
   # Entry point for the Insecure Hash Algorithm and Random tests
   class InsecureAlgorithmController < ApplicationController
     SOME_HARDCODED_PASSWORD = 'BadPracticeOfStoringPasswordInRepo'
-    SOME_HARDCODED_KEY = 'BadPracticeOfStoringKeyInRepo'
+    SOME_HARDCODED_KEY = [1,2,3]
```
Comment (srushe) [inline app/controllers/vulneruby_engine/insecure_algorithm_controller.rb:7]: Why this change? I'm intrigued :)

File: app/models/vulneruby_engine/secret_mongo.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+module VulnerubyEngine
+    class SecretMongo
+        include Mongoid::Document
+        field :name, type: String
+        field :value, type: String
+        field :created_at, :type => Date
+        field :updated_at, :type => Date
+    end
+end
```
Comment (srushe) [inline app/models/vulneruby_engine/secret_mongo.rb:11]:
```suggestion
module VulnerubyEngine
  class SecretMongo
    include Mongoid::Document
    field :name, type: String
    field :value, type: String
    field :created_at, :type => Date
    field :updated_at, :type => Date
  end
end
```

File: spec/dummy/config/application.rb [hunk 5]
```diff
@@ -14,7 +15,9 @@
 Bundler.require(:thin_min) if !!ENV['THIN_MIN']
 Bundler.require(:unicorn_min) if !!ENV['UNICORN_MIN']
 Bundler.require(:unicorn_max) if !!ENV['UNICORN_MAX']
+Bundler.require(:mongoid)
 
+# Mongoid.load!('/app/spec/dummy/config/mongoid.yml')
```
Comment (srushe) [inline spec/dummy/config/application.rb:20]: Can we just remove this?

**Review by srushe — APPROVED** [review]
Some questions and suggestions, but nothing blocking. LGTM.

File: app/controllers/vulneruby_engine/insecure_algorithm_controller.rb (hunk 3 above)
Comment (kaloyan-mm) [reply app/controllers/vulneruby_engine/insecure_algorithm_controller.rb:7]: Well according to our tests in the agent - we may have different values here to trigger hardcode-key so I switched it to trigger it, as the previous one wasn't

File: spec/dummy/config/application.rb (hunk 5 above)
Comment (kaloyan-mm) [reply spec/dummy/config/application.rb:20]: Yup

**Review by cshep-contrast — APPROVED** [review]

**Review by hmcginnis — APPROVED** [review]
